libavfilter/dnn/dnn_backend_common.c

```diff
@@ -101,14 +106,19 @@ DNNReturnType ff_dnn_async_module_cleanup(DNNAsyncExecModule *async_module)
 DNNReturnType ff_dnn_start_inference_async(void *ctx, DNNAsyncExecModule *async_module)
 {
     int ret;
+    void *status = 0;
 
     if (!async_module) {
         av_log(ctx, AV_LOG_ERROR, "async_module is null when starting async inference.\n");
         return DNN_ERROR;
     }
 
 #if HAVE_PTHREAD_CANCEL
-    pthread_join(async_module->thread_id, NULL);
+    pthread_join(async_module->thread_id, &status);
+    if (status == DNN_ASYNC_FAIL) {
```

Comment:
man pthread_join:
On success, pthread_join() returns 0; on error, it returns an error number.

can we just compare it with -1?

Reply:
The error number is when it cannot join the thread. In our case, the thread will join successfully and we are catching the status of `async_thread_routine()`. If we change the value of `DNN_ASYNC_SUCCESS` to something like `(void *)40`, then on the success the status will be 40 but the return value of `pthread_join()` will be 0. 

For example, the output of the below code is `0 -1`.

```C
void *runner(void *arg)
{
    sleep(1);
    return (void *)-1;
}

int main() {
    pthread_t thread;
    int returnValue = pthread_create(&thread, NULL, runner, NULL);
    void *status = 0;
    int ret = pthread_join(thread, &status);
    printf("%d %d\n", ret, (int *)status);
}
```

Reply:
thanks, i see, looks good.

Reply:
Just one question, do we need to check the exit status in `ff_dnn_async_module_cleanup` also? It is called only when the request item is destroyed to ensure the thread was joined.

Reply:
yes, i think so, we may output a warning here

Reply:
Okay, I'll add this to the patch. Thank you.